components/toggle/Toggle.js

```diff
@@ -1,28 +1,61 @@
-import React from 'react';
+import React, { useState } from 'react';
 import PropTypes from 'prop-types';
-import { t } from 'ttag';
+import { c } from 'ttag';
 
 import Input from '../input/Input';
-import Label from '../label/Label';
+import Icon from '../icon/Icon';
 
-const Toggle = ({ id }) => {
+const label = (type, key) => {
+    const I18N = {
+        on: c('Toggle button').t`On`,
+        off: c('Toggle button').t`Off`
+    };
+
+    if (type === 'icon') {
+        return (
+            <span className="pm-toggle-label-text">
+                <Icon name={key} alt={I18N[key]} className="pm-toggle-label-img" />
+            </span>
+        );
+    }
+
+    if (key === 'on') {
+        return <span className="pm-toggle-label-text">{I18N[key]}</span>;
+    }
+
+    return <span className="pm-toggle-label-text">{I18N[key]}</span>;
+};
+
+const Toggle = ({ id, checked, onChange, type }) => {
+    const [value, setValue] = useState(checked);
     return (
         <>
-            <Input type="checkbox" id={id} className="pm-toggle-checkbox" />
-            <Label htmlFor={id} className="pm-toggle-label">
-                <span className="pm-toggle-label-text">{t`On`}</span>
-                <span className="pm-toggle-label-text">{t`Off`}</span>
-            </Label>
+            <Input
+                type="checkbox"
+                id={id}
+                checked={value}
+                className="pm-toggle-checkbox"
+                onChange={() => onChange(value)}
+            />
+            <label htmlFor={id} className="pm-toggle-label" onClick={() => setValue(!value)}>
```

Comment:
Since you use a `label` connected with the input checkbox by the `id`, I don't think you need to define `onClick` attribute here. The `onChange` on `<Input>` is enough.